carma-messenger-ui/website/widgets/emergencyResponse/widget.js

```diff
@@ -0,0 +1,361 @@
+            let destTr = document.createElement('tr');
+            let destThRouteName = document.createElement('th');
+            destThRouteName.textContent = "Route Name";
+            destTr.appendChild(destThRouteName);
+            destHeader.appendChild(destTr);
+            destTbl.appendChild(destHeader);
+            let destBody = document.createElement('tbody');
+            let destTr1 = document.createElement('tr');
+            let destTrRouteName = document.createElement('th');
+            destTrRouteName.textContent = "NA";
+            destTrRouteName.id = "routeNameValue";
+            destTr1.appendChild(destTrRouteName);
+            destBody.appendChild(destTr1);
+            destTbl.appendChild(destBody);
+            destinationCol.appendChild(destTbl);
+            //Destination Button
+            let destBtn = document.createElement('button');
+            destBtn.className = "btn btn-danger dest-btn btn-lg";
+            destBtn.innerHTML = "Arrived at Emergency Location";
+            destBtn.setAttribute("title", "You will be redirected to event management page.");
+            destBtn.onclick = () => {
+                let success = service_arrive_at_emergency_destination();
+                if(success)
+                {
+                    goToEventManagement();
+                }                
+            };
+            destinationCol.appendChild(destBtn);
+            destinationRow.appendChild(destinationCol);
+
+            //Alert popup 
+            let alertRow = document.createElement('div');
+            alertRow.className = "row alert-row";
+            let alertCol = document.createElement('div');
+            alertCol.className = "col offset-sm-7";
+            alertCol.id = "alert-col";
+            let alert = createAlertDiv("");
+            alertCol.appendChild(alert);
+            alertRow.appendChild(alertCol);
+            container.appendChild(titleRow);
+            container.appendChild(vehicleStatusRow);
+            container.appendChild(destinationRow);
+            container.appendChild(alertRow);
+            $(this.element).append(container);
+        },
+        subscribe_bsm: function () {
+            subscribe_bsm();
+        },
+        service_get_emergency_route: function () {
+            service_get_emergency_route();
+        },
+        subscribe_alert: function () {
+            subscribe_alert();
+        },
+        _destroy: function () {
+            this.element.empty();
+            this._super();
+            console.log("Emergency Response configuration page is destroyed");
+        }
+    });//CarmaJS.
+
+    var loadCustomWidget = function (container) {
+        //create Emergency Response page
+        container.emergencyResponse();
+        container.emergencyResponse("subscribe_bsm", null);
+        container.emergencyResponse("subscribe_alert", null);
+        container.emergencyResponse("service_get_emergency_route", null);
```

Comment:
Does "service_arrive_at_emergency_destination" need to be listed here? Just checking

Reply:
https://github.com/usdot-fhwa-stol/carma-messenger/pull/172/commits/fa2f1d4ab5ee18dff527cd5e589b9c83508d0dd1

Reply:
Sorry, meant to say that I don't see "service_arrive_at_emergency_destination" listed here. Should it be?